travel-event-host-app/src/app/api/event/[id]/unregister/route.ts

```diff
@@ -0,0 +1,29 @@
+import { NextResponse } from 'next/server';
+import { connectMongoDB } from '@/lib/mongodb';
+import Event from '@/schemas/event';
+import mongoose from 'mongoose';
+
+export async function PATCH(req: Request, params: any) {
+  let { userId } = await req.json();
+
+  const id = params.id;
+  await connectMongoDB();
+  const isValidObjectId = mongoose.Types.ObjectId.isValid(id);
+  if (!isValidObjectId) {
+    return NextResponse.json({ message: 'Invalid ObjectId format' }, { status: 400 });
+  }
+
+  const eventFound = await Event.findById(id);
+
+  if (eventFound) {
+    eventFound.participantIds = eventFound.participantIds.filter(
+      (participant: { userId: string; timeStamp: Date }) => participant.userId !== userId,
```

Comment:
I think the logic would be more robust if it first checks that the user is in the event participant array. If they aren't, it should return an error response.